community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexAccessor.java

```diff
@@ -0,0 +1,149 @@
+    private final Selector selector;
+
+    FusionIndexAccessor( IndexAccessor nativeAccessor, IndexAccessor luceneAccessor, Selector selector )
+    {
+        this.nativeAccessor = nativeAccessor;
+        this.luceneAccessor = luceneAccessor;
+        this.selector = selector;
+    }
+
+    @Override
+    public void drop() throws IOException
+    {
+        try
+        {
+            nativeAccessor.drop();
+        }
+        finally
+        {
+            luceneAccessor.drop();
+        }
+    }
+
+    @Override
+    public IndexUpdater newUpdater( IndexUpdateMode mode )
+    {
+        return new FusionIndexUpdater( nativeAccessor.newUpdater( mode ), luceneAccessor.newUpdater( mode ), selector );
+    }
+
+    @Override
+    public void force() throws IOException
+    {
+        nativeAccessor.force();
+        luceneAccessor.force();
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        try
+        {
+            nativeAccessor.close();
+        }
+        finally
+        {
+            luceneAccessor.close();
+        }
+    }
+
+    @Override
+    public IndexReader newReader()
+    {
+        return new FusionIndexReader( nativeAccessor.newReader(), luceneAccessor.newReader(), selector );
+    }
+
+    @Override
+    public BoundedIterable<Long> newAllEntriesReader()
+    {
+        BoundedIterable<Long> nativeAllEntries = nativeAccessor.newAllEntriesReader();
+        BoundedIterable<Long> luceneAllEntries = luceneAccessor.newAllEntriesReader();
+        return new BoundedIterable<Long>()
+        {
+            @Override
+            public long maxCount()
+            {
+                long nativeMaxCount = nativeAllEntries.maxCount();
+                long luceneMaxCount = luceneAllEntries.maxCount();
+                return nativeMaxCount == UNKNOWN_MAX_COUNT || luceneMaxCount == UNKNOWN_MAX_COUNT ?
+                       UNKNOWN_MAX_COUNT : nativeMaxCount + luceneMaxCount;
+            }
+
+            @Override
+            public void close() throws Exception
+            {
+                try
+                {
+                    nativeAllEntries.close();
+                }
+                finally
+                {
+                    luceneAllEntries.close();
+                }
+            }
+
+            @Override
+            public Iterator<Long> iterator()
+            {
+                return Iterables.concat( nativeAllEntries, luceneAllEntries ).iterator();
+            }
+        };
+    }
+
+    @Override
+    public ResourceIterator<File> snapshotFiles() throws IOException
+    {
+        return concatResourceIterators(
+                asList( nativeAccessor.snapshotFiles(), luceneAccessor.snapshotFiles() ).iterator() );
+    }
+
+    @Override
+    public void verifyDeferredConstraints( PropertyAccessor propertyAccessor )
+            throws IndexEntryConflictException, IOException
+    {
+        nativeAccessor.verifyDeferredConstraints( propertyAccessor );
```

Comment:
just a note for question, not sure it actually will work as it is now for native index